core/src/main/kotlin/com/github/inflab/spring/data/mongodb/core/aggregation/ReplaceRootDsl.kt

```diff
@@ -0,0 +1,173 @@
+import org.springframework.data.mongodb.core.aggregation.ReplaceRootOperation
+import org.springframework.data.mongodb.core.aggregation.ReplaceRootOperation.ReplaceRootDocumentOperation
+import kotlin.reflect.KProperty
+
+/**
+ * A Kotlin DSL to configure $replaceRoot stage using idiomatic Kotlin code.
+ *
+ * @author username1103
+ * @since 1.0
+ * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot">$replaceRoot (aggregation)</a>
+ */
+@AggregationMarker
+class ReplaceRootDsl {
+    private var operation: ReplaceRootOperation? = null
+
+    /**
+     * Specifies the new root document with document operation.
+     *
+     * @param configuration The configuration for the [ReplacementDocumentOperationDsl]
+     */
+    fun newRoot(configuration: ReplacementDocumentOperationDsl.() -> Unit) {
+        operation = ReplacementDocumentOperationDsl().apply(configuration).get()
+    }
+
+    /**
+     * Specifies field name for the new root document.
+     *
+     * @param path The path of the field to replace new root.
+     */
+    fun newRoot(path: String) {
+        operation = ReplaceRootOperation.builder().withValueOf(path)
+    }
+
+    /**
+     * Specifies field name for the new root document.
+     *
+     * @param path The path of the field to replace new root.
+     */
+    fun newRoot(path: KProperty<*>) {
+        operation = ReplaceRootOperation.builder().withValueOf(path.toDotPath())
+    }
+
+    /**
+     * Specifies the new root document with aggregation expression.
+     *
+     * @param configuration The configuration block where you can use DSL to define aggregation expression.
+     */
+    fun expressions(configuration: AggregationExpressionDsl.() -> AggregationExpression) {
+        operation = ReplaceRootOperation.builder().withValueOf(AggregationExpressionDsl().configuration())
+    }
+
+    internal fun get() = checkNotNull(operation) { "ReplaceRoot operation must not be null!" }
+
+    /**
+     * A Kotlin DSL to configure replacement document using idiomatic Kotlin code.
+     */
+    class ReplacementDocumentOperationDsl(
+        private val prefix: String = "",
+        private var operation: ReplaceRootDocumentOperation = ReplaceRootDocumentOperation(),
+    ) {
+
+        /**
+         * Adds new fields to documents with aggregation expression.
+         *
+         * @param configuration The configuration block where you can use DSL to define aggregation expression.
+         */
+        infix fun String.set(configuration: AggregationExpressionDsl.() -> AggregationExpression) {
+            operation = operation.and(AggregationExpressionDsl().configuration()).`as`(this.addPrefix())
+        }
+
+        /**
+         * Adds new fields to documents from a field.
+         *
+         * @param path The path of the field to contain value to be added.
+         */
+        infix fun String.set(path: KProperty<Any?>) {
```

Comment:
we can take `Any?` to type parameter.
but It would be better to unify with one method.
```suggestion
        infix fun String.set(path: KProperty<*>) {
```